### PR TITLE
Fix Linux build. Require C99 & add POSIX macro

### LINKED ISSUE #5: Build breaks on old versions of GCC
I tried building on an old Linux machine running gcc 4.6.3 (2011) and GNU make 3.81 (2006) and ran into failures:

```sh
cc -c -g3 -ggdb -Werror -Wall -Wextra -Wformat-security -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wdisabled-optimization -Wfloat-equal -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpacked -Wpointer-arith -Wredundant-decls -Wstrict-prototypes -Wunreachable-code -Wwrite-strings -o md5.o md5.c
cc -c -g3 -ggdb -Werror -Wall -Wextra -Wformat-security -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wdisabled-optimization -Wfloat-equal -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpacked -Wpointer-arith -Wredundant-decls -Wstrict-prototypes -Wunreachable-code -Wwrite-strings -o sha256.o sha256.c
cc -g3 -ggdb -Werror -Wall -Wextra -Wformat-security -Waggregate-return -Wbad-function-cast -Wcast-align -Wdeclaration-after-statement -Wdisabled-optimization -Wfloat-equal -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpacked -Wpointer-arith -Wredundant-decls -Wstrict-prototypes -Wunreachable-code -Wwrite-strings -o nsfinfo nsfinfo.c md5.o sha256.o
nsfinfo.c: In function 'my_MD5_File':
nsfinfo.c:107:2: error: 'for' loop initial declarations are only allowed in C99 mode
nsfinfo.c:107:2: note: use option -std=c99 or -std=gnu99 to compile your code
nsfinfo.c: In function 'my_SHA256_File':
nsfinfo.c:163:2: error: 'for' loop initial declarations are only allowed in C99 mode
make: *** [nsfinfo] Error 1
```

I added `-std=c99` to the `CFLAGS` and then I got POSIX related failures instead:
```sh
nsfinfo.c: In function 'main':
nsfinfo.c:235:2: error: implicit declaration of function 'getopt' [-Werror=implicit-function-declaration]
nsfinfo.c:235:2: error: nested extern declaration of 'getopt' [-Werror=nested-externs]
nsfinfo.c:247:10: error: 'optind' undeclared (first use in this function)
nsfinfo.c:247:10: note: each undeclared identifier is reported only once for each function it appears in
cc1: all warnings being treated as errors
make: *** [nsfinfo] Error 1
```

### PR DESCRIPTION
**Pull request by ullenius**
Fixes #5 

* Add POSIX macro, see `man 7 feature_test_macros`
* Require C99
* Added `.POSIX:` to adhere to the makefile standard
> According to the standard, in order to get reliable POSIX behavior, the first non-comment line of the Makefile must be .POSIX.
https://nullprogram.com/blog/2017/08/20/

### CODE REVIEW

**Comment by ullenius** [comment]
An alternative solution is rewriting the `for`-loops using c89 (ANSI C) style:

```c
int i;
for (; i < 10; i++)
```

**Comment by koitsu** [comment]
This PR is really quite unhelpful for me.  Details:

1. The `.POSIX` declaration in a Makefile is nonsensical and nobody does this.
2. The only thing that I can see that is C99 are the new `my_{MD5,SHA256}_File` functions, where I declare a variable type within a `for()` loop.  I did this erroneously out of concern for `-Wunused`.  The solution is to simply declare `unsigned int i` at the top of the function/with the other variable definitions a la C89.  Please change your PR to do that instead and verify it works everywhere that you're testing it.
3. I have absolutely no idea what `#define _XOPEN_SOURCE 600` does.  `man 7 feature_test_macros` is only relevant on Linux, and a magic #define with a magic value does not tell me what this solves.

**Comment by ullenius** [comment]
Thanks for taking the time and effort to write a lengthy reply. Please bear in mind that I'm a C novice and thus I don't know how things are usually done.

**TL;DR** I'll close this PR. I will try your suggested approach instead on a new branch.

>1. The .POSIX declaration in a Makefile is nonsensical and nobody does this.

I have no strong opinion on this.

>2. Please change your PR to do that instead and verify it works everywhere that you're testing it.

I will try this and make a new PR if it works.

>3. I have absolutely no idea what #define _XOPEN_SOURCE 600

The `_XOPEN_SOURCE` feature test macro tells the compiler what version of POSIX you are using. This is as per the [standard](https://pubs.opengroup.org/onlinepubs/007904875/functions/xsh_chap02_02.html). I don't know if other compilers except GCC use it. The same thing can be achieved usinga  compiler flag:
 `cc -D_XOPEN_SOURCE           # Or any value < 500`

## References
[feature_test_macros(7) — Linux manual page](https://man7.org/linux/man-pages/man7/feature_test_macros.7.html)
